pkgs/applications/networking/instant-messengers/ripcord/default.nix

```diff
@@ -8,29 +8,44 @@ mkDerivation rec {
   version = "0.4.29";
 
   src = let
+    appbundle = fetchzip {
+      url = "https://cancel.fm/dl/Ripcord_Mac_${version}.zip";
+      sha256 = "sha256-v8iydjLBjFN5LuctpcBpEkhSICxPhLKzLjSASWtsQok=";
+      stripRoot = false;
+    };
+
     appimage = fetchurl {
       url = "https://cancel.fm/dl/Ripcord-${version}-x86_64.AppImage";
       sha256 = "sha256-4yDLPEBDsPKWtLwdpmSyl3b5XCwLAr2/EVtNRrFmmJk=";
       name = "${pname}-${version}.AppImage";
     };
-  in appimageTools.extract {
+  in if (stdenv.isDarwin) then appbundle else appimageTools.extract {
     name = "${pname}-${version}";
     src = appimage;
   };
 
-  nativeBuildInputs = [ autoPatchelfHook desktop-file-utils imagemagick ];
-  buildInputs = [ libsodium libopus libGL alsa-lib ]
-    ++ [ qtbase qtsvg qtmultimedia qtwebsockets qtimageformats ]
-    ++ (with xorg; [ libX11 libXScrnSaver libXcursor xkeyboardconfig ]);
+  nativeBuildInputs = lib.optionals stdenv.isLinux [
```

Comment:
With so many differences between darwin and linux, might as well define it in a separate mkDerivation call

Reply:
Sure